### PR TITLE
CLI: Cleanup create options

### PR DESCRIPTION
**Pull request by louib**
Fixes https://github.com/keepassxreboot/keepassxc/issues/4307

This makes sure that there's no clash between the database opening options (`--no-password` and `--key-file`) and the database creation/modification options, which start by `--set`. The main consequence is that setting the password on creation must be done explicitly, but I feel like it's ok to be explicit when creating the  db. 

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Refactor (significant modification to existing code)
- ✅ New feature (non-breaking change which adds functionality)
- ✅ Breaking change (fix or feature that would cause existing functionality to change)
- ✅ Documentation (non-code change)

## Testing strategy
unit tests.

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ My change requires a change to the documentation, and I have updated it accordingly.
- ✅ I have added tests to cover my changes.

### CODE REVIEW

File: src/cli/Create.cpp [hunk 1]
```diff
@@ -97,21 +109,24 @@ int Create::execute(const QStringList& arguments)
...
+        if (!password.isNull()) {
+            key->addKey(password);
+        }
     }
 
-    QSharedPointer<FileKey> fileKey;
-    if (parser->isSet(Command::KeyFileOption)) {
-        if (!loadFileKey(parser->value(Command::KeyFileOption), fileKey)) {
+    if (parser->isSet(Create::SetKeyFileOption)) {
+        QSharedPointer<FileKey> fileKey;
```
Comment (droidmonkey) [inline src/cli/Create.cpp]: Egad, the name swap is killing me here. Can we standardize on `keyFile` please.

File: src/cli/Create.cpp [hunk 2]
```diff
@@ -97,21 +109,24 @@ int Create::execute(const QStringList& arguments)
 
     auto key = QSharedPointer<CompositeKey>::create();
 
-    auto password = Utils::getPasswordFromStdin();
-    if (!password.isNull()) {
-        key->addKey(password);
+    if (parser->isSet(Create::SetPasswordOption)) {
+        auto password = Utils::getPasswordFromStdin();
+        if (!password.isNull()) {
```
Comment (droidmonkey) [inline src/cli/Create.cpp]: I prefer to use `QString::isEmpty()`.

**Comment by droidmonkey** [comment]
If neither `--set-password` nor `--set-keyfile` are provided we should add a prompt to confirm that the user intends to create a database without any protection. This prompt would not show if the `--quiet` flag is set.

**Comment by louib** [comment]
> If neither --set-password nor --set-keyfile are provided we should add a prompt to confirm that the user intends to create a database without any protection. This prompt would not show if the --quiet flag is set.

Can we create a database without any protection from the GUI?

File: src/cli/Create.cpp (hunk 1 above)
Comment (louib) [reply src/cli/Create.cpp]: all the keys are named like that (`PasswordKey`, `CompositeKey`, etc). You would rename the CLI option?

File: src/cli/Create.cpp (hunk 2 above)
Comment (louib) [reply src/cli/Create.cpp]: This is actually a `QSharedPointer<PasswordKey>`, and not a `QString` password. I renamed the variable to make it more obvious.

**Comment by droidmonkey** [comment]
Yes you can, but it gives you a warning

File: src/cli/Create.cpp (hunk 1 above)
Comment (droidmonkey) [reply src/cli/Create.cpp]: You can use fileKey

**Comment by louib** [comment]
@droidmonkey of course we could also preserve the current default of creating the db with a password. Not sure what is the best solution here.

**Comment by droidmonkey** [comment]
I'm confused, can't we just do:

```
if (!parser->isSet(Command::KeyFileOption) && !parser->isSet(Command::PasswordOption) && !parser->isSet(Command::NoCredentialsOption)) {
	err << QObject::tr("Cannot create database without any credentials. If this is what you intended to do, use --set-no-credentials.");
	return EXIT_FAILURE;
}
```

**Comment by louib** [comment]
@droidmonkey yeah, we could :100: do this.

**Comment by droidmonkey** [comment]
@louib are you going to complete this PR?

**Comment by louib** [comment]
@droidmonkey yes, I'm going to need a couple more weeks before I can jump back into it.

**Comment by droidmonkey** [comment]
I can close this one out to get it off the books, otherwise it might not make 2.6

**Comment by droidmonkey** [comment]
So you cannot create a database with no credentials in the GUI so I did not allow that behavior in CLI. I did, however, implement password repeat checks and the ability to create a database without a password. This applies to the `import` command as well.

**Comment by louib** [comment]
@droidmonkey so you decided not to implement a `--no-credentials` option? I don't mind the `Y/N` question, but I feel like it makes the process less suitable for scripting.

**Comment by droidmonkey** [comment]
That is only for a blank password. It is no less scriptable than having to enter a password in the first place.

**Pull request merged**